plugin/manager.go

```diff
@@ -109,22 +109,15 @@ func GetManager() *Manager {
 
 // Init (was NewManager) instantiates the singleton Manager.
 // TODO: revert this to NewManager once we get rid of all the singletons.
-func Init(root, execRoot string, remote libcontainerd.Remote, rs registry.Service, liveRestore bool) (err error) {
+func Init(root string, remote libcontainerd.Remote, rs registry.Service, liveRestore bool) (err error) {
 	if manager != nil {
 		return nil
 	}
 
 	root = filepath.Join(root, "plugins")
-	execRoot = filepath.Join(execRoot, "plugins")
-	for _, dir := range []string{root, execRoot} {
-		if err := os.MkdirAll(dir, 0700); err != nil {
-			return err
-		}
-	}
```

Comment:
@thaJeztah : 

1. parameterizing runRoot: there's no value add in writing a unit-test for runRoot sockets. Integration tests will catch any issues with the socket.

2. "should we move the creation of /run/docker/ (now at line 128) here, to keep the same behavior as before (create the path before constructing the manager) "
No. runRoot dir creation is same as before. i.e after manager object was created. It perfectly fine this way.

3. "Also looks like we're no longer creating the root/plugins directory now."
No, we are. The os.MkdirAll() call later on creates it,
https://github.com/docker/docker/blob/master/plugin/backend.go#L62